src/Microsoft.TestPlatform.Execution.Shared/DebuggerBreakpoint.cs

```diff
@@ -0,0 +1,71 @@
+﻿using Microsoft.VisualStudio.TestPlatform.PlatformAbstractions;
+using Microsoft.VisualStudio.TestPlatform.Utilities;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.VisualStudio.TestPlatform.Execution
+{
+    internal static class DebuggerBreakpoint
+    {
+        internal static void WaitForDebugger(string environmentVariable)
+        {
+            if (string.IsNullOrWhiteSpace(environmentVariable))
+            {
+                throw new ArgumentException($"'{nameof(environmentVariable)}' cannot be null or whitespace.", nameof(environmentVariable));
+            }
+
+            var debugEnabled = Environment.GetEnvironmentVariable(environmentVariable);
+            if (!string.IsNullOrEmpty(debugEnabled) && debugEnabled.Equals("1", StringComparison.Ordinal))
+            {
+                ConsoleOutput.Instance.WriteLine("Waiting for debugger attach...", OutputLevel.Information);
+
+                var currentProcess = Process.GetCurrentProcess();
+                ConsoleOutput.Instance.WriteLine(
+                    string.Format("Process Id: {0}, Name: {1}", currentProcess.Id, currentProcess.ProcessName),
+                    OutputLevel.Information);
+
+                while (!Debugger.IsAttached)
```

Comment:
This line sometimes hand forever. (Especially when the user doesn't know that they are debugging.) Should we try attaching to a debugger using `Debugger.Attach`?

Reply:
I don't think so. In most cases you don't have anything to attach to, or you don't want to start VS for every start of the executable.

Reply:
I think it doesn't start a new instance of VS. It will show this dialog from which you can an select an existing instance or launch a new one.

![image](https://user-images.githubusercontent.com/433724/126503450-bb224745-bf14-4c33-a2f9-1c0945270154.png)

Reply:
I just moved the code, I don't want to also change the behavior in this PR. If you want to do it, please start a new PR. I will give it a try.